Configuração do pre-commit com ruff e Atualização do Readme

sarahcalixto opened the pull request:
#3

sarahcalixto committed:
Configuração do pre-commit com ruff e Atualização do Readme

thiag0bezerra committed:
Delete README.md

thiag0bezerra reviewed (changes requested):
Muito bem. Quase lá.

Falta apenas realmente registrar o `pre-commit` como dependência do projeto

Sugiro que execute `uv add --dev pre-commit`, isso adicionará a última versão da ferramenta ao `pyproject.toml` e ao `uv.lock`

Para instalar é necessário executar `uv run pre-commit install`. Isso vai gerar uma mensagem como:

```txt
pre-commit installed at .git\hooks\pre-commit
```

Em seguida faremos o teste definitivo (simulando um commit com) com `uv run pre-commit run --all-files`. Isso vai gerar algo tipo:

```txt
[INFO] Initializing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Installing environment for https://github.com/astral-sh/ruff-pre-commit.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
ruff check...............................................................Passed
ruff format..............................................................Failed
- hook id: ruff-format
- files were modified by this hook
```

Não tem problema se aparecer "Failed" aqui. Basicamente ele estará dizendo que nem tudo estava formatado (mas agora estará).

Por fim é esperado que os arquivos `pyproject.toml` e `uv.lock` estejam pendentes no git. Basta adicioná-los, fazer o commit e submetê-los a essa mesma branch.

---

Importante. Garanta que você está na branch correta antes de realizar as operações acima.

sarahcalixto committed:
chore(devtools): adiciona pre-commit ao grupo de desenvolvimento

Inclui pre-commit (>=4.2.0) no grupo dev do pyproject.toml e atualiza o lockfile para garantir reprodutibilidade do ambiente.

sarahcalixto committed:
Merge branch 'resolucao-issues' of https://github.com/cacdia/lattes into resolucao-issues

thiag0bezerra reviewed (approved):
LGTM
